plugin/storage/es/spanstore/reader.go

```diff
@@ -129,10 +129,17 @@ type SpanReaderParams struct {
 
 // NewSpanReader returns a new SpanReader with a metrics.
 func NewSpanReader(p SpanReaderParams) *SpanReader {
+	maxSpanAge := p.MaxSpanAge
+	// If rollover is enabled set lookback far into the past
+	// In rollover only read alias is used to query the data so looking far into the past should
+	// not affect performance.
+	if p.UseReadWriteAliases {
+		maxSpanAge = time.Hour * 24 * 365 * 100
```

Comment:
I think we should extend `TestNewSpanReader`, maybe converting it to a table-driven test, to prevent a regression on this functionality.

Reply:
+1